src/common/crypto_format.c

```diff
@@ -162,6 +162,44 @@ curve25519_public_from_base64(curve25519_public_key_t *pkey,
   }
 }
 
+/** Encode <b>skey</b> as a base64-encoded string, with trailing "="
+ * characters, in the buffer <b>output</b>, which must have at least
+ * CURVE25519_BASE64_PADDED_LEN+1 bytes available. Return 0 on success, -1 on
+ * failure. */
+int
+curve25519_secret_to_base64(char *output,
+                            const curve25519_secret_key_t *skey)
+{
+  char buf[128];
+  base64_encode(buf, sizeof(buf),
+                (const char*)skey->secret_key, CURVE25519_SECKEY_LEN, 0);
+  buf[CURVE25519_BASE64_PADDED_LEN] = '\0';
+  memcpy(output, buf, CURVE25519_BASE64_PADDED_LEN+1);
+  return 0;
+}
+
+/** Try to decode a base64-encoded curve25519 secret key from <b>input</b>
+ * into the object at <b>skey</b>. Return 0 on success, -1 on failure.
+ * Accepts keys with or without a trailing "=". */
+int
+curve25519_secret_from_base64(curve25519_secret_key_t *skey,
+                              const char *input)
+{
+  size_t len = strlen(input);
+  if (len == CURVE25519_BASE64_PADDED_LEN - 1) {
+    /* not padded */
+    return digest256_from_base64((char*)skey->secret_key, input);
+  } else if (len == CURVE25519_BASE64_PADDED_LEN) {
+    char buf[128];
+    if (base64_decode(buf, sizeof(buf), input, len) != CURVE25519_SECKEY_LEN)
+      return -1;
+    memcpy(skey->secret_key, buf, CURVE25519_SECKEY_LEN);
+    return 0;
+  } else {
+    return -1;
+  }
+}
```

Comment:
There is probably a time we will want to merge those with the public key as well since they have the same exact code except with different objects. But we can do that later.